aosp_diff/preliminary/hardware/interfaces/09_0009-Add-the-memtrack-AIDL-HAL-implement.patch

```diff
@@ -0,0 +1,206 @@
+From 940b50ee2d4ba3f0a347bf359ab20302e97ab4e7 Mon Sep 17 00:00:00 2001
+From: HeYue <yue.he@intel.com>
+Date: Wed, 18 May 2022 12:34:36 +0800
+Subject: [PATCH] Add the memtrack AIDL HAL implement
+
+Tracked-On: OAM-98918
+Signed-off-by: HeYue <yue.he@intel.com>
+---
+ memtrack/aidl/default/Android.bp          |  3 +
+ memtrack/aidl/default/Memtrack.cpp        | 73 ++++++++++++++++++++++-
+ memtrack/aidl/default/Memtrack.h          | 15 +++++
+ memtrack/aidl/default/main.cpp            |  5 +-
+ memtrack/aidl/default/memtrack-default.rc |  2 +-
+ 5 files changed, 94 insertions(+), 4 deletions(-)
+
+diff --git a/memtrack/aidl/default/Android.bp b/memtrack/aidl/default/Android.bp
+index 7a7feea5bfa3..5fd86f1c3881 100644
+--- a/memtrack/aidl/default/Android.bp
++++ b/memtrack/aidl/default/Android.bp
+@@ -30,6 +30,9 @@ cc_binary {
+     shared_libs: [
+         "libbase",
+         "libbinder_ndk",
++        "liblog",
++        "libhardware",
++        "libutils",
+         "android.hardware.memtrack-V1-ndk_platform",
+     ],
+     srcs: [
+diff --git a/memtrack/aidl/default/Memtrack.cpp b/memtrack/aidl/default/Memtrack.cpp
+index 49a6582989ca..1c4c3fcf94bb 100644
+--- a/memtrack/aidl/default/Memtrack.cpp
++++ b/memtrack/aidl/default/Memtrack.cpp
+@@ -14,12 +14,28 @@
+  * limitations under the License.
+  */
+ 
++#define LOG_TAG "memtrack"
++
+ #include "Memtrack.h"
+ 
++#include <log/log.h>
++
++#include <hardware/hardware.h>
++#include <hardware/memtrack.h>
++
+ namespace aidl {
+ namespace android {
+ namespace hardware {
+ namespace memtrack {
++namespace impl {
++namespace example {
```

Comment:
why need the namespace example?

Reply:
i checked the power and light aidl implementation, they all use the impl and example name space.
The memtrack service name is android.hardware.memtrack-service.example
So i add it there.

Reply:
AOSP provide the example, but our implementation should not be example anymore.

Reply:
The light aidl hal is implemented by Yuanzhe before, he also use the example, you could check here:
https://github.com/projectceladon/vendor-intel-utils/pull/1070/files
I could remove the example namespace, i just follow the other aild implementation examples.